extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js

```diff
@@ -642,10 +642,8 @@ export class AmpLightboxGallery extends AMP.BaseElement {
    */
   openLightboxForElement_(element) {
     this.currentElemId_ = element.lightboxItemId;
-    // Hack to access private property. Better than not getting
-    // type checking to work.
-    /**@type {?}*/ (this.carousel_).implementation_.showSlideWhenReady(
-        this.currentElemId_);
+    dev().assert(this.carousel_).getImpl()
```

Comment:
@alanorozco note I updated all uses of `_implementation` for `<amp-carousel>` in this component as these are easy to test. I'll send a follow up PR for switching out `_implementation` for accessing internals of `<amp-image-viewer>` because the changing that to a promise involves more refactoring.

Reply:
Awesome, thanks for taking care of our codebase 🌱

Reply:
Just saw this, this is awesome @cathyxz !